assets/js/components/GoogleChart.js

```diff
@@ -20,205 +20,251 @@
-		};
-	}, [] );
+	if ( onSelect ) {
+		combinedChartEvents.push( {
+			eventName: 'select',
+			callback: onSelect,
+		} );
+	}
 
 	return (
-		<div className="googlesitekit-graph-wrapper">
-			<div ref={ chartRef } className="googlesitekit-line-chart">
-				<div className="googlesitekit-chart-loading">
-					{ loading && (
-						<div className="googlesitekit-chart-loading__wrapper">
-							{ loadText && (
-								<p>{ __( 'Loading chart…', 'google-site-kit' ) }</p>
-							) }
-
-							<ProgressBar
-								className={ className }
-								small={ loadSmall }
-								compress={ loadCompressed }
-								height={ loadHeight }
-							/>
-						</div>
-					) }
-				</div>
-			</div>
+		<div
+			className={ classnames(
+				'googlesitekit-chart',
+				`googlesitekit-chart--${ chartType }`,
+				className
+			) }
+		>
+			<Chart
+				className="googlesitekit-chart__inner"
+				chartEvents={ combinedChartEvents }
+				chartType={ chartType }
+				chartVersion="49"
+				data={ modifiedData }
+				loader={ loader }
+				height={ height }
+				getChartWrapper={ ( chartWrapper, google ) => {
+					// Remove all the event listeners on the old chart before we draw
+					// a new one. Only run this if the old chart and the new chart aren't
+					// the same reference though, otherwise we'll remove existing `onReady`
+					// events and other event listeners, which will cause bugs.
+					if ( chartWrapper !== chartWrapperRef.current ) {
+						// eslint-disable-next-line no-unused-expressions
+						googleRef.current?.visualization.events.removeAllListeners( chartWrapperRef.current?.getChart() );
+						// eslint-disable-next-line no-unused-expressions
+						googleRef.current?.visualization.events.removeAllListeners( chartWrapperRef.current );
+					}
+
+					chartWrapperRef.current = chartWrapper;
+					googleRef.current = google;
+
+					if ( getChartWrapper ) {
+						getChartWrapper( chartWrapper, google );
+					}
+				} }
+				width={ width }
+				{ ...otherProps }
+			/>
+			{ children }
 		</div>
 	);
 }
 
 GoogleChart.propTypes = {
-	chartID: PropTypes.string,
-	chartType: PropTypes.oneOf( [ 'area', 'pie', 'line' ] ).isRequired,
 	className: PropTypes.string,
-	data: PropTypes.arrayOf( PropTypes.array ),
-	loadCompressed: PropTypes.bool,
-	loadSmall: PropTypes.bool,
-	loadHeight: PropTypes.number,
+	children: PropTypes.node,
+	chartEvents: PropTypes.arrayOf( PropTypes.shape( {
+		eventName: PropTypes.string,
+		callback: PropTypes.func,
+	} ) ),
+	chartType: PropTypes.oneOf( [
+		'AnnotationChart',
+		'AreaChart',
+		'BarChart',
+		'BubbleChart',
+		'Calendar',
+		'CandlestickChart',
+		'ColumnChart',
+		'ComboChart',
+		'DiffChart',
+		'DonutChart',
+		'Gantt',
+		'Gauge',
+		'GeoChart',
+		'Histogram',
+		'LineChart',
+		'Line',
+		'Bar',
+		'Map',
+		'OrgChart',
+		'PieChart',
+		'Sankey',
+		'ScatterChart',
+		'SteppedAreaChart',
+		'Table',
+		'Timeline',
+		'TreeMap',
+		'WaterfallChart',
+		'WordTree',
+	] ).isRequired,
```

Comment:
My only worry there is not allowing valid types might be confusing for future developers who will think our charts can't/shouldn't use those types…

But I'll update it and add a comment that lets them know more can be added, that seems fine… in case we want to add any special handling for other charts before allowing their types, like we do with loaders here.